### PR TITLE
Add a preprocessor for the Mistral backbone

### PR DESCRIPTION
**Pull request by tirthasheshpatel**
This PR adds a preprocessor for the `keras_nlp.models.Mistral` backbone.

### CODE REVIEW

File: keras_nlp/models/mistral/mistral_preprocessor.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+from keras_nlp.utils.keras_utils import (
+    convert_inputs_to_list_of_tensor_segments,
+)
+from keras_nlp.utils.keras_utils import pack_x_y_sample_weight
+from keras_nlp.utils.python_utils import classproperty
+
+
+@keras_nlp_export("keras_nlp.models.MistralPreprocessor")
+class MistralPreprocessor(Preprocessor):
+    """An Mistral preprocessing layer which tokenizes and packs inputs.
```
Comment (mattdangerw) [inline keras_nlp/models/mistral/mistral_preprocessor.py:28]: An Mistral -> A Mistral

File: keras_nlp/models/mistral/mistral_preprocessor_test.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        )
+
+    def test_errors_for_2d_list_input(self):
+        preprocessor = MistralPreprocessor(**self.init_kwargs)
+        ambiguous_input = [["one", "two"], ["three", "four"]]
+        with self.assertRaises(ValueError):
+            preprocessor(ambiguous_input)
+
+    @pytest.mark.extra_large
+    def test_all_presets(self):
```
Comment (mattdangerw) [inline keras_nlp/models/mistral/mistral_preprocessor_test.py:64]: let's save the preset testing until we actually do presets

**Review by mattdangerw — APPROVED** [review]
Looks good! Just a couple comments

File: keras_nlp/models/mistral/mistral_preprocessor.py (hunk 1 above)
Comment (tirthasheshpatel) [inline keras_nlp/models/mistral/mistral_preprocessor.py:28]:
```suggestion
    """A Mistral preprocessing layer which tokenizes and packs inputs.
```

File: keras_nlp/models/mistral/mistral_preprocessor_test.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        for preset in MistralPreprocessor.presets:
+            self.run_preset_test(
+                cls=MistralPreprocessor,
+                preset=preset,
+                input_data=self.input_data,
+            )
```
Comment (tirthasheshpatel) [inline keras_nlp/models/mistral/mistral_preprocessor_test.py:70]:
```suggestion
            preprocessor(ambiguous_input)
```

File: keras_nlp/models/mistral/mistral_preprocessor_test.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+import pytest
```
Comment (tirthasheshpatel) [inline keras_nlp/models/mistral/mistral_preprocessor_test.py:17]:
```suggestion
import os
```

**Comment by tirthasheshpatel** [comment]
All tests passing, so merging. Will propose the 7B preset next.

**Pull request merged**